### PR TITLE
Jazz API_MUNCHER

### PR DESCRIPTION
**Pull request by jazziesf**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
How did you go about exploring the Edamam API, and how did you try querying the API? | I used postman to build the api query by testing the syntax out on the application. 
What is an API Wrapper? Why is it in `lib`? How would your project change if you needed to interact with more than one API (aka more than just the Edamam API)? | An API wrapper isolates all interaction with other servers by using a wrapper.  Create another API wrapper to manage the request separately. 
Describe your API wrapper, the methods you created in it, and why you decided to create those methods/how they were helpful | The index displays the the search result acquired by the API wrapper from Edamam database and displays up to 100 recipes.  The instance variable limits the number of items shown per page up to 100 recipes. 
What was an edge case or failure case test you wrote for your API Wrapper? What was a nominal case? | Nominal case It returns an empty array if query is blank. Return recipe show page for a single recipe. 
How does VCR aid in testing an API? | It stores API test that could help you limit the number of request you make by recording test data. 
What is the Heroku URL of your deployed application? | https://git.heroku.com/api-mucherada.git

### CODE REVIEW

File: test/controllers/recipes_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    must_respond_with :ok
+  end
+
+  it "with redirect to the home page for an invalid uri" do
+    VCR.use_cassette('invalid_uri_file') do
+      id = "9b5945e03f0"
+
+      get recipes_path(id)
+      
+      value(response).must_be :successful?
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:53]: You don't seem to be testing a redirect here.

File: app/controllers/recipes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class RecipesController < ApplicationController
+require 'will_paginate/array'
+
+  def index
+    recipes = EdamamApiWrapper.list_recipes(params[:query])
+    @recipes = recipes.paginate(:page => params[:page], :per_page => 10)
+  end
+
+  def show
+    if @recipe.nil?
```
Comment (CheezItMan) [inline app/controllers/recipes_controller.rb:11]: You haven't tried to find the recipe first, so of course it will be `nil`.

So this show page **will not work**.

File: test/lib/edamam_api_wrapper_test.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  #   end
+  # end
+
+  it 'will return a single recipe' do
+    VCR.use_cassette('single_recipe') do
+      single_recipe = EdamamApiWrapper.find_recipe("460c55360ede96cf3afca927baefd1de")
+
+      expect(single_recipe).must_be_instance_of Recipe
+    end
+  end
```
Comment (CheezItMan) [inline test/lib/edamam_api_wrapper_test.rb:32]: You also **must** have a negative case here.

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Core Requirements** |
Git hygiene |  Not very many commits, but it was a smaller project.  Overall good commit messages
Comprehension questions |  Check, I think you labeled the edge-case as nominal in one of your answers. 
**General** |
Rails fundamentals (RESTful routing, use of named paths) |  `recipes/new` really isn't quite right as it's not making a new recipe.  Instead I would have called it `recipes/search` or `recipes/index` (just making the search page the index page.  
Semantic HTML |  A little bit of a forest of divs here.  
Errors are reported to the user |  Check
API Wrapper to handle the API requests |  Check, I like how you broke out the ID in the search functionality.  
Controller testing |  Check
Lib testing |  You are missing a negative case for the `show` action.  You need to test where the ID is wrong.  
Search Functionality |  Check
List Functionality |  Check
Show individual item functionality | Corrected, nice work
**Styling** |
List view shows 10 items at a time and/or has pagination |  Check
Attractive and usable UI |  Very nice UI, very impressive
**API Features** |
The App attributes Edamam |  Check
The VCR cassettes do not contain the API key |Check
**External Resources** |
Link to deployed app on Heroku |  **Your heroku link isn't correct**, I found it [here](http://api-mucherada.herokuapp.com).
**Overall** |  Please look at your `show` action in your controller and resubmit.  Right now it's not at standard due to a small mistake there, which I think you could correct pretty quickly.

File: app/controllers/recipes_controller.rb (hunk 2 above)
Comment (CheezItMan) [reply app/controllers/recipes_controller.rb:11]: Correction, this will now work.  Good update
